### PR TITLE
Improved enemies scripts

### PR DESCRIPTION
**Pull request by Rubetoman**
Improved the enemies behaviour:
- Now every enemy uses the nav mesh to move.
- Cleaned unnecessary code and added some comments.
- Fixed some bugs.

To test:
- Make sure to recompile all the scripts before testing.
- Load "EnemiesTest" scene.
- Generate the nav mesh if not already.
- Press play and check that the enemy behaviour works.
- Check that the enemies avoid the holes of the nav mesh.

### CODE REVIEW

File: Scripts/EnemyControllerScript/EnemyControllerScript.h [hunk 1]
```diff
@@ -43,9 +43,9 @@ class EnemyControllerScript_API EnemyControllerScript : public Script
 
 	inline bool IsCollidingWithPlayer() const;
 
-	void Move(float speed, math::float3& direction) const;
+	void Move(float speed, math::float3& direction) const;		// To be deprecated: old method, doesn't use nav mesh
```
Comment (AlfonsoMestres) [inline Scripts/EnemyControllerScript/EnemyControllerScript.h]: Avoid using comments to mark a function as Deprecated, instead, use this above your function to be deprecated:

`[Obsolete("Function is deprecated, please use Function 2 instead.")]`
`public void Function()
{
... stuff
}`

This way people will get notified when trying to use the function in their Visual Studio.

File: Scripts/EnemyControllerScript/EnemyControllerScript.h [hunk 2]
```diff
@@ -43,9 +43,9 @@ class EnemyControllerScript_API EnemyControllerScript : public Script
...
 	void Move(float speed, float& refreshTime, math::float3 position, std::vector<float3>& path) const; // Move using nav mesh
-	void MoveTowards(float speed) const;
+	void MoveTowards(float speed) const;						// To be deprecated: old method, doesn't use nav mesh
```
Comment (AlfonsoMestres) [inline Scripts/EnemyControllerScript/EnemyControllerScript.h]: Avoid using comments to mark a function as Deprecated, instead, use this above your function to be deprecated:

`[Obsolete("Function is deprecated, please use Function 2 instead.")]`
`public void Function()
{
... stuff
}`

This way people will get notified when trying to use the function in their Visual Studio.

**Comment by AlfonsoMestres** [comment]
EnemyUsingNavMesh

Working! Nice job 💯

**Comment by Gabroide** [comment]
Hero doesn't move, just attack movement!

**Pull request merged**